### PR TITLE
Readme updates 

### PR DESCRIPTION
**Pull request by mskarlin**
Prepping the new readme to capture all the recent work and re-introduce PaperQA.

Added a little baby change for an edge case that I found via parsing tons of documents -- sometimes weak LLMs will try to add a list of authors that's empty to the docs object after peeking.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,40 +29,117 @@ Tulevski2007: Tulevski, George S., et al. "Chemically assisted directed assembly
...
 
+## Install
+
+To use the full suite of features in PaperQA, you need to install it with the optional `agents` extra:
+
+```bash
+pip install paper-qa[agents]
+```
+
+PaperQA uses an LLM to operate, so you'll need to either set an appropriate [API key environment variable](https://docs.litellm.ai/docs/providers) (i.e. `export OPENAI_API_KEY=sk-...`) or set up an open source LLM server (i.e. using [ollama](https://github.com/ollama/ollama)). Any LiteLLM compatible model can be configured to use with PaperQA.
```
Comment (jamesbraza) [inline README.md:40]: Can you mention `litellm` here

File: README.md [hunk 2]
```diff
@@ -44,40 +29,117 @@ Tulevski2007: Tulevski, George S., et al. "Chemically assisted directed assembly
...
+
+If you need to index a large set of papers (100+), you will likely want an API key for both [Crossref](https://www.crossref.org/documentation/metadata-plus/metadata-plus-keys/) and [Semantic Scholar](https://www.semanticscholar.org/product/api#api-key), which will allow you to avoid hitting public rate limits using these metadata services. Those can be exported as `CROSSREF_API_KEY` and `SEMANTIC_SCHOLAR_API_KEY` variables.
```
Comment (jamesbraza) [inline README.md:42]: Should we mention Google Scholar here?

File: README.md [hunk 3]
```diff
@@ -44,40 +29,117 @@ Tulevski2007: Tulevski, George S., et al. "Chemically assisted directed assembly
...
+
 ## What's New?
 
-Version 4 removed langchain from the package because it no longer supports pickling. This also simplifies the package a bit - especially prompts. Langchain can still be used, but it's not required. You can use any LLMs from langchain, but you will need to use the `LangchainLLMModel` class to wrap the model.
+Version 5 added a CLI, agentic workflows, and removed much of the state from the `Docs` object. `Docs` objects pickled from prior versions of `PaperQA` are not compatible with version 5 and will need to be rebuilt.
```
Comment (jamesbraza) [inline README.md:46]: Should we keep v4 here and just add v5?

File: README.md [hunk 4]
```diff
@@ -44,40 +29,117 @@ Tulevski2007: Tulevski, George S., et al. "Chemically assisted directed assembly
...
+
+### Module Usage
+
+PaperQA's full workflow can be accessed via Python directly:
+
+```Python
+from paperqa Settings
+from paperqa.agents import ask
+
+answer = ask('What manufacturing challenges are unique to bispecific antibodies?', settings=Settings(temperature=0.5))
```
Comment (jamesbraza) [inline README.md:96]: Wdyt of adding `blacken-docs`

File: README.md [hunk 5]
```diff
@@ -44,40 +29,117 @@ Tulevski2007: Tulevski, George S., et al. "Chemically assisted directed assembly
...
+
+```bash
+pqa --parsing.chunk_size 5000 ask 'What manufacturing challenges are unique to bispecific antibodies?'
+```
+
+### Module Usage
+
+PaperQA's full workflow can be accessed via Python directly:
+
+```Python
```
Comment (jamesbraza) [inline README.md:92]: Normally it's `python` (lowercase)

File: README.md [hunk 6]
```diff
@@ -421,20 +444,24 @@ def make_typewriter(step_name):
...
 ## Customizing Prompts
 
 You can customize any of the prompts, using the `PromptCollection` class. For example, if you want to change the prompt for the question, you can do:
 
 ```python
-from paperqa import Docs, Answer, PromptCollection
+from paperqa import (
+    Docs,
+    Settings,
+)
```
Comment (jamesbraza) [inline README.md:463]:
```suggestion
from paperqa import Docs, Settings
```

File: README.md [hunk 7]
```diff
@@ -445,8 +472,13 @@ my_qaprompt = (
...
     "Context: {context}\n\n"
 )
-prompts = PromptCollection(qa=my_qaprompt)
-docs = Docs(prompts=prompts)
+
+docs = Docs()
+
+docs.query(
+    "Are covid-19 vaccines effective?",
+    settings=Setting(prompts=PromptSettings(qa=my_qaprompt)),
```
Comment (jamesbraza) [inline README.md:480]: Can we name this `my_qa_prompt`

File: paperqa/utils.py [hunk 8]
```diff
@@ -383,11 +383,14 @@ def bibtex_field_extract(
...
-    author_rep = (
-        author[0].split()[-1].casefold()
-        if "Unknown" not in author[0]
-        else "unknownauthors"
-    )
+    try:
+        author_rep = (
+            author[0].split()[-1].casefold()
+            if "Unknown" not in author[0]
+            else "unknownauthors"
```
Comment (jamesbraza) [inline paperqa/utils.py:390]: Wdyt of moving `unknownauthors` to a module level variable

**Review by jamesbraza — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mskarlin) [reply README.md:40]: more than the last sentence?

File: README.md (hunk 3 above)
Comment (mskarlin) [reply README.md:46]: Nah, I think that "what's new" implies the most recent change, the other stuff can come from release notes.

**Pull request merged**